### PR TITLE
15863 removed the shift of radio buttons in approval and restoration …

### PR DESCRIPTION
**Pull request by PaulGarewal**
…type

*Issue #:* /bcgov/entity#15863

*Description of changes:*

- The radio buttons in Approval Type and Restoration Type shift when the field is invalid (or left blank). The changes removes that shift by adjusting the margins of the invalid section. 

image

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the bcrs-entities-create-ui license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-create-dev--pr-571-zlaa1beo.web.app

File: src/components/Restoration/ApprovalType.vue [hunk 1]
```diff
@@ -3,6 +3,7 @@
     <div
       class="section-container"
       :class="{ 'invalid-section': invalidSection }"
+      :style="invalidSection ? {margin: '0 9px 0 0'} : {}"
```
Comment (JazzarKarim) [inline src/components/Restoration/ApprovalType.vue:6]: Paul, we usually don't add in-line styling. Would it be possible to add a conditional class for this maybe?

Comment (PaulGarewal) [reply src/components/Restoration/ApprovalType.vue:6]: Of course, I will change that right away.

Comment (severinbeauvais) [reply src/components/Restoration/ApprovalType.vue:6]: Unless you need exactly 9px of right margin, you can use a [Vuetify spacing helper class](https://v2.vuetifyjs.com/en/styles/spacing/) here. In this case, `mr-2`.

Comment (severinbeauvais) [reply src/components/Restoration/ApprovalType.vue:6]: By the way,  you can have multiple dynamic classes. Something like this:
```
:class="{
  'myClass1': myCondition1,
  'myClass2': myCondition2
}"
```

**Review by severinbeauvais — COMMENTED** [review]
How does adding right margin fix the issue of the whole block being moved to the right by the width of the error border?

File: src/components/Restoration/ApprovalType.vue (hunk 1 above)
Comment (PaulGarewal) [reply src/components/Restoration/ApprovalType.vue:6]: So much easier, learning the power of Vue

**Comment by PaulGarewal** [comment]
> How does adding right margin fix the issue of the whole block being moved to the right by the width of the error border?

It does act like a squeeze, as that fixes the radio buttons in place, which was the issue. However, I can change the logic here by just making it red and not shifting at all (which is what you proposed). I will just have to find out where that is in the code.

**Comment by severinbeauvais** [comment]
> > How does adding right margin fix the issue of the whole block being moved to the right by the width of the error border?
> 
> It does act like a squeeze, as that fixes the radio buttons in place, which was the issue. However, I can change the logic here by just making it red and not shifting at all (which is what you proposed). I will just have to find out where that is in the code.

We try to consistently use top whitespace (not bottom) and left whitespace (not right), which makes it easier to lay things out than if we mixed top/bottom and left/right. I think your solution could cause extra work in the future if anything needs to be re-aligned, so might as well fix it consistently now.

Yes, I propose either a permanent left border, which is only red when needed, or show the red border but then pull that component left by the same amount (with -ve margin).

**Review by severinbeauvais — APPROVED** [review]
LGTM

**Comment by severinbeauvais** [comment]
Please fix the lint errors. You can ignore the lint warnings unless you caused them ;)

**Review by JazzarKarim — APPROVED** [review]
LGTM Paul! 👍

**Pull request merged**